### PR TITLE
feat: added flag to preserve asset uid in the referenced entries when new asset version published

### PR DESCRIPTION
**Pull request by aman19K**
- [x] Updating an asset version renders it null in referenced entries until they are republished (CS-42732)
- [x] npm audit fix
- [x] version bump
- [x] rimraf version update
- [x] remove sast-scan

### CODE REVIEW

File: package.json [hunk 1]
```diff
@@ -43,16 +43,16 @@
     "debug": "^4.3.4",
     "lodash": "^4.17.15",
     "mkdirp": "^1.0.4",
-    "rimraf": "^3.0.2",
+    "rimraf": "^5.0.5",
```
Comment (antonyagustine) [inline package.json:46]: Hi @aman19K this package updated to a MAJOR version. Have we validated the breaking changes of the respective package and our use case?

File: src/fs.ts [hunk 2]
```diff
@@ -557,9 +557,12 @@ export class FilesystemStore {
...
         }
-        const entries = assetMap[asset.uid];
-        for (const entry of entries) {
-          await this._updateEntryAssetReference(entry, asset.uid);
+        //NOTE:- Preserve asset uid in the referenced entries instead of updating it with null, When a new asset version is published.
+        if(!this.config.preserveAssetInReferencedEntries){
+          const entries = assetMap[asset.uid];
+          for (const entry of entries) {
+            await this._updateEntryAssetReference(entry, asset.uid);
+          }
```
Comment (antonyagustine) [inline src/fs.ts:565]: Have we validated this config through the DataSync boilerplate?

File: package.json (hunk 1 above)
Comment (aman19K) [reply package.json:46]: Yes @antonyagustine , we're only using it for removing dist folder.

File: src/fs.ts (hunk 2 above)
Comment (aman19K) [reply src/fs.ts:565]: @antonyagustine Yes, preserveAssetInReferencedEntries= true then we're not updating referenced entries with null. By default existing functionality.

**Pull request merged**